src/oni2-keyboard-layout/Oni2_KeyboardLayout.re

```diff
@@ -0,0 +1,45 @@
+type callback = unit => unit;
+
+let callbackListRef: ref(list(callback)) = ref([]);
+
+let subscribe = (callback: callback) => {
+  callbackListRef := [callback, ...callbackListRef^];
+
+  () =>
+    callbackListRef := List.filter(cb => cb !== callback, callbackListRef^);
+};
+
+external init: unit => unit = "oni2_KeyboardLayoutInit";
+
+external getCurrentLayout: unit => string =
+  "oni2_KeyboardLayoutGetCurrentLayout";
+external getCurrentLanguage: unit => string =
+  "oni2_KeyboardLayoutGetCurrentLanguage";
+
+Callback.register("oni2_KeyboardLayoutCallbackListRef", callbackListRef);
```

Comment:
Here, the C binding could be made simpler if we have a wrapper function like:

```
let callAllKeyboardCallbacks = () => {
 callbackListRef^ |> List.iter(f => f());
};
Callback.register("oni2_KeyboardLayoutCallbackListRef", callAllKeyboardCallbacks);
```
Means less juggling to do in the C bindings, which is nice

Reply:
That's a much better idea! I changed it to this in dc53ea4. Let me know if that's better!